src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -51,44 +54,119 @@ public Class<CreateCollectionCommand> getCommandClass() {
 
   @Override
   public Operation resolveCommand(CommandContext ctx, CreateCollectionCommand command) {
-    if (command.options() != null && command.options().vector() != null) {
-      if (!dataStoreConfig.vectorSearchEnabled()) {
-        throw new JsonApiException(
-            ErrorCode.VECTOR_SEARCH_NOT_AVAILABLE,
-            ErrorCode.VECTOR_SEARCH_NOT_AVAILABLE.getMessage());
+
+    if (command.options() != null) {
+      String vectorize = null;
+      String indexing = null;
+      int vectorSize = 0;
+      String function = null;
+
+      // handling indexing options
+      if (command.options().indexing() != null) {
```

Comment:
Would it make more sense to move this validation in `IndexingConfig` instance itself? Like `validate()` method or something -- this would encapsulate logic bit better.
Or maybe better yet, add `create()` method in `IndexingConfig` to encapsulate these, validate during construction?

Also: I think `allow` and `deny` are always both non-null since they are initialized as empty `Set`s?

Reply:
Moved it inside IndexingConfig. validate(). Have provided `@Nullable` for both allow and deny. So will be null in case they are not provided in the request.

Reply:
> I think `allow` and `deny` are always both non-null since they are initialized as empty `Set`s

For cache I have different`IndexingConfig` inside `CollectionSetting` these fields are Set in there. Also we have not used the Command object in other places.

Reply:
I don't think `@Nullable` annotation will inject anything (they are mostly just for documentation?). But since tests pass, I guess check is ok.

But ok. I missed separate `IndexingConfig`; if they are different I guess it's ok.

Reply:
Ok 2 instances of `IndexingConfig` makes sense, all good. :)